src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -703,7 +703,7 @@ private function updateData(array $data, string $attribute, mixed $attributeValu
     private function isMaxDepthReached(array $attributesMetadata, string $class, string $attribute, array &$context): bool
     {
         if (!($enableMaxDepth = $context[self::ENABLE_MAX_DEPTH] ?? $this->defaultContext[self::ENABLE_MAX_DEPTH] ?? false)
-            || null === $maxDepth = $attributesMetadata[$attribute]?->getMaxDepth()
+            || !isset($attributesMetadata[$attribute]) || null === $maxDepth = $attributesMetadata[$attribute]?->getMaxDepth()
```

Comment:
Could be:

```suggestion
            || null === $maxDepth = ($attributesMetadata[$attribute]?->getMaxDepth() ?? null)
```

Reply:
The warning is still thrown in that case as `$attributesMetadata[$attribute]` is still trying to be accessed.